configure.ac

```diff
@@ -787,6 +790,46 @@ AS_IF([test x$with_lustre != xno],[
        MOTR_LINUX_SENDFILE
 ])
 
+#
+# Checking libfabric library----------------------------------------------- {{{1
+#
+
+AC_ARG_ENABLE([libfab],
+        [AS_HELP_STRING([--enable-libfab],
+                        [Use libfabric transport])],
+        [],[enable_libfab=yes]
+)
+
+AC_ARG_WITH([libfab],
+        [AS_HELP_STRING([--with-libfab=path],
+                        [set path to user installed libfabric headers and library])],
+        [LIBFABRIC=$with_libfab],
+        [LIBFABRIC=""]
+)
+
+AS_IF([! test -z "$with_libfab"],
+      [
+        LIB_FAB_INCLUDES="-I$LIBFABRIC/include"
+        LIBFAB_LIBS="-L$LIBFABRIC/lib -lfabric"
+        M0_CPPFLAGS="-DENABLE_LIBFAB $M0_CPPFLAGS $LIB_FAB_INCLUDES"
+      ],
+)
+AS_IF([test x$enable_libfab = xyes],
+      [
+	MOTR_SEARCH_LIBS([fi_fabric], [fabric], [LIBFAB_LIBS],
+                  [fi_fabric cannot be found! Try to install libfabric  package]
+          )
+	AC_CHECK_HEADERS([rdma/fabric.h], [],
+		AC_MSG_ERROR([rdma/fabric.h is not found. Try to install libfabric package])
+          )
+        M0_CPPFLAGS="-DENABLE_LIBFAB $M0_CPPFLAGS"
+      ]
+)
+
+AC_SUBST([LIBFAB_LIBS])
+AM_CONDITIONAL([ENABLE_LIBFAB],
+               [test "x$enable_libfab" = xyes || ! test -z "$with_libfab"])
+
```

Comment:
Please make sure and test that,
* If libfabric is available, we will use libfabric. Otherwise, we will not use it.
  -  we will not report error if libfab is not available.
* If  "--enable-libfab" or "--with-libfab" provided, we are forced to use libfabric. If libfabric is not available, we will report error.

Reply:
We had enabled libfabric by default in this branch for the custom build. 
We have updated it now as per the above mentioned functionality.